internal/enginecache/cache.go

```diff
@@ -190,21 +190,28 @@ func downloadFile(filepath string, url string) error {
 	<-doneCompletedCh         // wait for signal that printing has completed
 
 	elapsed := time.Since(start)
-	log.Printf("\033[2K\rDownload completed in %s\n", elapsed)
-
+	log.Printf("\033[2K\rhover: Download completed in %.2fs\n", elapsed.Seconds())
 	return nil
 }
 
-// ValidateOrUpdateEngine validates the engine we have cached matches the
-// flutter version, or otherwise downloads a new engine. The returned path is
-// that of the engine location.
-func ValidateOrUpdateEngine(targetOS string) (engineCachePath string) {
-	engineCachePath = filepath.Join(cachePath(), "hover", "engine", targetOS)
+// ValidateOrUpdateEnginePath validates the engine we have cached matches the
+// flutter version, or otherwise downloads a new engine. The engine cache
+// location is set by the the user.
+func ValidateOrUpdateEnginePath(targetOS string, cachePath string) (engineCachePath string) {
```

Comment:
`ValidateOrUpdateEnginePath` sounds like it updates the EnginePath. Perhaps rename to `ValidateOrUpdateEngineAtPath`